.github/workflows/codeql.yml

```diff
@@ -18,6 +18,8 @@ on:
   pull_request:
     branches:
       - main
+    paths-ignore:
```

Comment:
This will exclude it only for PR not on main. You can do https://stackoverflow.com/a/78835420